internal/model/live_update.go

```diff
@@ -5,16 +5,31 @@ import (
 )
 
 // Specifies how to update a running container.
+// 0. If any paths specified in a FallBackOn step have changed, fall back to an image build
+//    (i.e. don't do a LiveUpdate)
 // 1. If there are Sync steps in `Steps`, files will be synced as specified.
 // 2. Any time we sync one or more files, all Run and RestartContainer steps will be evaluated.
 type LiveUpdate struct {
-	Steps []LiveUpdateStep
-
-	// When files matching any of these paths change, we should fall back to a full rebuild.
-	FullRebuildTriggers PathSet
+	Steps   []LiveUpdateStep
+	BaseDir string // directory where the LiveUpdate was initialized (we'll use this to eval. any relative paths)
 }
 
-func NewLiveUpdate(steps []LiveUpdateStep, fullRebuildTriggers PathSet) (LiveUpdate, error) {
+func NewLiveUpdate(steps []LiveUpdateStep, baseDir string) (LiveUpdate, error) {
+	// Check that all FallBackOn steps come at the beginning
+	// (Technically could do this in the loop below, but it's
```

Comment:
happy to hear that i should just do all this checking in one big loop -- my reasoning here is that if we did all order validation in a single loop, any time we add a new step, we'll have to remember to set seenNonFallBackStep when we check them. But i don't have strong feelings 🤷‍♀️

Reply:
this is probably fine for now, but this might get unwieldy as we get more restrictions. it might be better to have an explicit state machine